receiver/pulsarreceiver/README.md

```diff
@@ -17,49 +17,78 @@ Pulsar receiver receives logs, metrics, and traces from Pulsar.
 ## Getting Started
 
 The following settings can be optionally configured:
+
 - `endpoint` (default = pulsar://localhost:6650): The url of pulsar cluster.
-- `topic` (default = otlp_spans for traces, otlp_metrics for metrics, otlp_logs for logs): The name of the pulsar topic to consume from.
-- `encoding` (default = otlp_proto): The encoding of the payload sent to pulsar. Available encodings:
-    - `otlp_proto`: the payload is deserialized to `ExportTraceServiceRequest`.
-    - `jaeger_proto`: the payload is deserialized to a single Jaeger proto `Span`.
-    - `jaeger_json`: the payload is deserialized to a single Jaeger JSON Span using `jsonpb`.
-    - `zipkin_proto`: the payload is deserialized into a list of Zipkin proto spans.
-    - `zipkin_json`: the payload is deserialized into a list of Zipkin V2 JSON spans.
-    - `zipkin_thrift`: the payload is deserialized into a list of Zipkin Thrift spans.
-- `consumer_name`: specifies the consumer name.
+- `trace`
```

Comment:
Why do we need to make this breaking change if we still want to migrate to the encoding extensions after that? I think it's better to avoid introducing this intermediary config interface.

Also, we need to try to avoid such breaking changes. Ideally, we need to keep the old interface functioning for some time with deprecation warning logs.

If you have any other changes not breaking the user config like refactoring, feel free to submit other PRs.

Reply:
The introduction of new encoding methods is only a small part of the reason, I created an issue for the PR https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28685, please take a look.

The component currently in the Alpha phase, I think the configuration changes comply with the agreement.

Reply:
@dmitryax

Reply:
Even for alpha, we should try to avoid breaking changes if possible.